### DISCUSSION TITLE
report error "No module named" when use ssh engine to debug multiple file

### DISCUSSION THREAD

**Discussion by wilikeit** [pyscripter/pyscripter#1325] [Q&A] [answered]
pyscripter 5.0.1

main.py
![image](https://github.com/user-attachments/assets/b4bb67d1-1986-481e-821c-0449c0b0f8ae)

fun1.py
```
def abc():
  print('hi')
```

**If I not set as shown in the picture below, the ssh engine will not automatically copy the project directory demo to the remote, need to manually copy demo to the remote？** 
I found that ：When Iuse the ssh engine to debug a single file, you do not need to go through the "file→Menu →open remote file", it seems that the single file will be automatically copied to the remote directory, Which remote directory is it copied to?

![image](https://github.com/user-attachments/assets/8c2ad6fc-a80a-422a-9de8-6d068a7e8d69)

**Comment by pyscripter** [comment, marked as answer]
> need to manually copy demo to the remote?

Yes

> it seems that the single file will be automatically copied to the remote directory, Which remote directory is it copied to?

No it is not copied.  The source code of the editor file is compiled directly and then is executed.  This is the case with the Remote Python engine as well.   The python engine is configured as if it was running a source file on disk.

So the proper way is to have your python source files on the remote server, open them in PyScripter using "Open Remote File" and then run/debug them.

**Reply by wilikeit** [reply]
Thanks for the answer. Just like I expected,Originally so, but this is not a problem, if I use a virtual machine development, can also directly mount the local directory to the virtual machine, automatic synchronization.
